pzsd_bot/__main__.py

```diff
@@ -1,48 +1,214 @@
+handler = logging.FileHandler(filename="pzsd_bot.log", encoding="utf-8", mode="a")
+handler.setFormatter(
+    logging.Formatter("%(asctime)s:%(levelname)s:%(name)s: %(message)s")
+)
 logger.addHandler(handler)
 
-bot = discord.Bot()
+bot = discord.Bot(intents=Intents.all())
+
+point_pattern = re.compile(
+    r"(?:^| )(?P<point_amount>[+-]?(?:\d+|\d{1,3}(?:,\d{3})*)) "
+    r"+points? (?:to|for) (?:(?P<recipient_name>\w+)|<@(?P<recipient_id>\d+)>)",
+    re.IGNORECASE,
+)
 
 
 @bot.event
 async def on_ready():
-    print(f"We have logged in as {bot.user}")
+    print(f"Ready.")
+    logger.info("Logged in as %s", bot.user)
 
 
 @bot.event
 async def on_message(message):
     if message.author == bot.user:
         return
 
+    if match := point_pattern.search(message.content):
+        point_amount = int(match["point_amount"].replace(",", ""))
+        recipient_name = match["recipient_name"]
+        recipient_id = match["recipient_id"]
+
+        async with engine.connect() as conn:
+            result = await conn.execute(
+                select(pzsd_user).where(
+                    pzsd_user.c.discord_snowflake == str(message.author.id)
+                )
+            )
+            try:
+                bestower = result.one_or_none()
+            except MultipleResultsFound:
+                logger.error(
+                    "Multiple users with snowflake '%s' found in user table",
+                    message.author.id,
+                )
+                raise
+
+            if bestower is None:
+                logger.info(
+                    "User '%s' with snowflake '%s' tried to bestow points but wasn't in the user table",
+                    message.author.name,
+                    message.author.id,
+                )
+                return
+
+            if recipient_name:
+                result = await conn.execute(
+                    select(pzsd_user).where(pzsd_user.c.name == recipient_name.lower())
+                )
+            else:
+                result = await conn.execute(
+                    select(pzsd_user).where(
+                        pzsd_user.c.discord_snowflake == recipient_id
+                    )
+                )
+            try:
+                recipient = result.one_or_none()
+            except MultipleResultsFound:
+                recipient_type = "name" if recipient_name else "snowflake"
+                logger.error(
+                    "Multiple users with %s '%s' found in user table",
+                    recipient_type,
+                    recipient_name or recipient_id,
+                )
+                raise
+
+            if recipient is None:
+                logger.info(
+                    "%s tried to bestow points to '%s' but they weren't in the user table",
+                    bestower.name,
+                    recipient_name or recipient_id,
+                )
+                return
+
+        excessive_point_violation = abs(point_amount) > POINT_MAX_VALUE
```

Comment:
-9223372036854775808 is a valid bigint

Reply:
But POINT_MAX_VALUE is 9223372036854775807

Reply:
dammit you're right